youtube_dl/extractor/snapchat.py

```diff
@@ -0,0 +1,64 @@
+# coding: utf-8
+from __future__ import unicode_literals
+import re
+
+from ..utils import unified_timestamp, parse_duration
+from .common import InfoExtractor
+
+
+class SnapchatIE(InfoExtractor):
+    _VALID_URL = r'https?://(?:www\.)?story.snapchat\.com/o/(?P<id>\w+)'
+    _TEST = {
+        'url': 'https://story.snapchat.com/o/W7_EDlXWTBiXAEEniNoMPwAAYuz9_1mcdex8MAXndPyIOAXndPyFyAO1OAA',
+        'md5': 'ab1900981cadcd955aae32a526096cbd',
+        'info_dict': {
+            'id': 'W7_EDlXWTBiXAEEniNoMPwAAYuz9_1mcdex8MAXndPyIOAXndPyFyAO1OAA',
+            'ext': 'mp4',
+            'title': 'W7_EDlXWTBiXAEEniNoMPwAAYuz9_1mcdex8MAXndPyIOAXndPyFyAO1OAA',
+            'thumbnail': 'https://s.sc-cdn.net/p9mgs8YyMZzfFRls4HGZxwXWBGVF-cEa6ZYPuCT2xPk=/default/preview.jpg',
+            'description': '#spotlight',
+            'timestamp': 1622914559,
+            'upload_date': '20210605',
+            'view_count': 62100,
+            'uploader_id': None
+        }
+    }
+
+    def _real_extract(self, url):
+        video_id = self._match_id(url)
+        webpage = self._download_webpage(url, video_id)
+
+        schema_video_object_raw = self._html_search_regex(r'<script data-react-helmet="true" type="application/ld\+json">(.+)</script>',
+                                                          webpage, 'schema_video_object')
+        schema_video_object = self._parse_json(schema_video_object_raw, video_id, fatal=True)
+        video_url = schema_video_object.get('contentUrl')
```

Comment:
Strictly, the JSON may not have resulted in a `schema_video_object` that is a `dict`, even if that case is an error, and even if it is the required `video_url` might not be found. You could consider, eg
```
        try:
            video_url = str_or_none(schema_video_object['contentUrl'])
            if not video_url:
                raise ValueError('video_url must be non-empty string')
        except (TypeError, ValueError) as e:
            raise ExtractorError('Unexpected format for schema_video_object', cause=e, video_id=video_id)
```